app/controllers/review_answers_controller.rb

```diff
@@ -0,0 +1,35 @@
+class ReviewAnswersController < ApplicationController
+  before_action :authenticate_user!
+
+  respond_to :json
+
+  def create_or_update
+    @review = Review.find(params[:review_id])
+    p = answer_params
+    @answer = @review.answers.where(review_question_id: p[:review_question_id]).first
+    unless @answer
```

Comment:
Just a bit of consideration on style: it takes me several brain cycles to understand what `unless/else` does every time, sorry :grin: If you could swap them over, I think it would make it more straightforward to read. 

Alternatively, `@answer = @review.answers.create_with(p).find_or_create_by(review_question_id: p[:review_question_id])` could be used to avoid the conditional block